test/async-hooks/test-async-local-storage-socket.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const net = require('net');
+const { AsyncLocalStorage } = require('async_hooks');
+
+net
+  .createServer((socket) => {
+    socket.write('Hello, world!');
+    socket.pipe(socket);
```

Comment:
What does it do?

Reply:
@orgads this creates a mock server for the first test case you shared in the issue.

Reply:
I meant specifically this line.

```javascript
socket.pipe(socket);
```

Does it echo whatever the socket accepts back to the client?